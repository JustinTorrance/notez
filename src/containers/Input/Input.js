import React, { Component } from 'react';

class Input extends Component {
  constructor() {
    super()
    this.state = {
      text: '',
      completed: false
    }
  }

  handleChange = (e) => {
    this.setState({ text: e.target.value })
  }

  handleBlur = (e) => {
    if (e.target.value.length > 0) {
      this.props.createNewItem(this.state.text)
    }
  }

  handleChecked = (e) => {
    const value = e.target.type
    if (value === 'checkbox') {
      this.setState({completed: !this.state.completed})
    }
  }

  render() {
    return (
      <div>
        <input
          name="completed"
          type="checkbox"
          checked={this.state.completed}
          onChange={this.handleChecked}
        />
        <input
          placeholder='description'
          value={this.state.text}
          onChange={this.handleChange}
          onBlur={this.handleBlur}
        />
      </div>
    );
  }
}

export default Input;