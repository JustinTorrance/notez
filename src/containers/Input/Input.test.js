import React from 'react';
import { shallow } from 'enzyme' ;
import Input from './Input.js';

describe('Input', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Input/>)
    expect(wrapper).toMatchSnapshot()
  })
})