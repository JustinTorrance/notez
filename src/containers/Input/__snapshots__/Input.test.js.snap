// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match the snapshot 1`] = `
"<div>
  <input className=\\"new-note-inputs\\" placeholder=\\"description\\" value=\\"\\" onChange={[Function]} onBlur={[Function]} />
</div>"
`;
